### ISSUE TITLE
static labels on MetericsFactory not added

### ISSUE THREAD

**Issue by cubakid** [prometheus-net/prometheus-net#389]
Given the following code, I expect 3 different counters each with a value of 1. However, it appears that a single counter is increased 3 times.

```
using ConsoleApp1;
using Prometheus;

Console.WriteLine( "Hello, World!" );

Metrics.SuppressDefaultMetrics();

var metricServer = new KestrelMetricServer( port: 1234 );
metricServer.Start();

Metrics.DefaultRegistry.SetStaticLabels( new Dictionary<string, string>
	{
	  // Labels applied to all metrics in the registry.
	  { "DefaultRegistry", "testing" }
	}
);

var c1 = new Class1( "c1" );
var c2 = new Class1( "c2" );
var c3 = new Class1( "c3" );

Console.ReadKey();
```
```
using Prometheus;

namespace ConsoleApp1
{
	internal class Class1
	{
		private IMetricFactory MetricFactory { get; set; }

		private Counter c1 { get; set; }

		public Class1( string name )
		{
			// Labels applied to all metrics created via this factory.
			MetricFactory = Metrics.WithLabels( new Dictionary<string, string> { { "testname", name } } );

			c1 = MetricFactory.CreateCounter( "test", "test counter" );

			c1.Inc();
		}
	}
}
```

# results
![image](https://user-images.githubusercontent.com/6548616/202864444-1bec8841-dd2c-4ac1-8a5b-58738e3a4079.png)

**Comment by cubakid** [comment]
Did some debugging. I'd rather leave it to the experts for a PR, but it seems that in `CollectorRegistry.GetOrAdd`, when creating the `CollectorIdentity`, the StaticLables.Names should be passed instead of the Values. That generated what I expected to see and matched the documentation.

![image](https://user-images.githubusercontent.com/6548616/202870729-c21bac1b-c571-4f32-afd3-5531485c4c58.png)

**Comment by cubakid** [comment]
While the output above seems to scrape by Prometheus OK, it goes against the sorting and group rules here [https://prometheus.io/docs/instrumenting/exposition_formats/#grouping-and-sorting](url)

**Comment by sandersaares** [comment]
Thanks for reporting! The fix will be in 8.0.0.
